pkg/nto/performanceprofile_test.go

```diff
@@ -604,6 +609,145 @@ func TestPerformanceProfileWithWorkloadHints(t *testing.T) {
 	}
 }
 
+func TestPerformanceProfileWithAnnotations(t *testing.T) {
+	testCases := []struct {
+		testAnnotations   map[string]string
+		expectedErrorText string
+	}{
+		{
+			testAnnotations:   map[string]string{"performance.openshift.io/ignore-cgroups-version": "true"},
+			expectedErrorText: "",
+		},
+		{
+			testAnnotations: map[string]string{
+				"kubeletconfig.experimental": "{'systemReserved':{'cpu':'500m','memory':'28Gi'}}",
+			},
+			expectedErrorText: "",
+		},
+		{
+			testAnnotations: map[string]string{"performance.openshift.io/ignore-cgroups-version": "true",
+				"kubeletconfig.experimental": "{\"systemReserved\":{\"cpu\":\"500m\",\"memory\":\"28Gi\"}}"},
+			expectedErrorText: "",
+		},
+		{
+			testAnnotations:   map[string]string{},
+			expectedErrorText: "'annotations' argument cannot be empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject())
+
+		result := testBuilder.WithAnnotations(testCase.testAnnotations)
+
+		if testCase.expectedErrorText != "" {
+			assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testAnnotations, result.Definition.ObjectMeta.Annotations)
+		}
+	}
+}
+
+//nolint:funlen
+func TestPerformanceProfileWithNet(t *testing.T) {
+	testCases := []struct {
+		testUserLevelNet  bool
```

Comment:
Do we want any test with testUserLevelNet set to false? All the tests are true.

Reply:
ups, one sec adding